### PR TITLE
[android] Read NFC tag containing CHIP setup payload

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Read CHIP setup payload encoded as URI NDEF Record with
scheme "ch:". This allows to use exactly the same setup
payload as for QR Code generation.

After a succesful NFC Tag read automatically switch to
the Device Details screen of the app to present decoded
information.

### CODE REVIEW

File: src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/CHIPToolActivity.kt [hunk 1]
```diff
@@ -89,6 +96,31 @@ class CHIPToolActivity :
...
+  private fun onNfcIntent(intent: Intent?) {
+    // Require 1 NDEF message containing 1 NDEF record
+    val messages = intent?.getParcelableArrayExtra(NfcAdapter.EXTRA_NDEF_MESSAGES)
+    if (messages?.size != 1) return;
+
+    val records = (messages[0] as NdefMessage).records
+    if (records.size != 1) return;
+
+    // Require NDEF URI record starting with "ch:"
+    val uri = records[0].toUri()
```
Comment (LuDuda) [inline src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/CHIPToolActivity.kt:108]: What happens with spaces, especially when they appear at the end of the string?

File: src/android/CHIPTool/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -23,6 +24,11 @@
                 <action android:name="android.intent.action.MAIN" />
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
+            <intent-filter>
+                <action android:name="android.nfc.action.NDEF_DISCOVERED" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <data android:scheme="ch" /><!-- Process CHIP URIs -->
```
Comment (LuDuda) [inline src/android/CHIPTool/app/src/main/AndroidManifest.xml:30]: How to ensure, that other organization won't take "CH" scheme? I guess we (Zigbee Alliance) register it in IANA, right?

**Review by LuDuda — APPROVED** [review]

File: src/android/CHIPTool/app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (Damian-Nordic) [reply src/android/CHIPTool/app/src/main/AndroidManifest.xml:30]: I don't think Apple registered their scheme "X-HM:", so not sure CHIP will do that either, but yes, schemes can be registered in IANA according to wikipedia ;)

File: src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/CHIPToolActivity.kt (hunk 1 above)
Comment (Damian-Nordic) [reply src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/CHIPToolActivity.kt:108]: Seems to be working correctly, but I agree we should bring that up during spec review (which I saw you had already done).

**Pull request closed without merging**